tests/v2/validation/provisioning/rke2/rke2_encryption_key_rotation_test.go

```diff
@@ -0,0 +1,190 @@
+package rke2
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+
+	rkev1 "github.com/rancher/rancher/pkg/apis/rke.cattle.io/v1"
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	"github.com/rancher/rancher/tests/framework/extensions/cloudcredentials"
+	"github.com/rancher/rancher/tests/framework/extensions/clusters"
+	"github.com/rancher/rancher/tests/framework/extensions/machinepools"
+	"github.com/rancher/rancher/tests/framework/pkg/config"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+	"github.com/rancher/rancher/tests/framework/pkg/wait"
+	"github.com/rancher/rancher/tests/integration/pkg/defaults"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
+)
+
+type RKE2EncryptionKeyRotationTestSuite struct {
+	suite.Suite
+	session     *session.Session
+	client      *rancher.Client
+	config      *Config
+	clusterName string
+	namespace   string
+}
+
+var phases = []rkev1.RotateEncryptionKeysPhase{
+	rkev1.RotateEncryptionKeysPhasePrepare,
+	rkev1.RotateEncryptionKeysPhasePostPrepareRestart,
+	rkev1.RotateEncryptionKeysPhaseRotate,
+	rkev1.RotateEncryptionKeysPhasePostRotateRestart,
+	rkev1.RotateEncryptionKeysPhaseReencrypt,
+	rkev1.RotateEncryptionKeysPhasePostReencryptRestart,
+	rkev1.RotateEncryptionKeysPhaseDone,
+}
+
+func (r *RKE2EncryptionKeyRotationTestSuite) TearDownSuite() {
+	r.session.Cleanup()
+}
+
+func (r *RKE2EncryptionKeyRotationTestSuite) SetupSuite() {
+	testSession := session.NewSession(r.T())
+	r.session = testSession
+
+	r.config = new(Config)
+	config.LoadConfig(ConfigurationFileKey, r.config)
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(r.T(), err)
+
+	r.client = client
+
+	r.clusterName = r.client.RancherConfig.ClusterName
+	r.namespace = r.client.RancherConfig.ClusterName
+}
+
+func (r *RKE2EncryptionKeyRotationTestSuite) TestEncryptionKeyRotationImpl(provider Provider, kubeVersion string, nodesAndRoles []machinepools.NodeRoles, credential *cloudcredentials.CloudCredential) {
+	name := fmt.Sprintf("Provider_%s/Kubernetes_Version_%s/Nodes_%v", provider.Name, kubeVersion, nodesAndRoles)
+	r.Run(name, func() {
+		testSession := session.NewSession(r.T())
+		defer testSession.Cleanup()
+
+		testSessionClient, err := r.client.WithSession(testSession)
+		require.NoError(r.T(), err)
+
+		clusterName := AppendRandomString(fmt.Sprintf("%s-%s", r.clusterName, provider.Name))
+		generatedPoolName := fmt.Sprintf("nc-%s-pool1-", clusterName)
+		machinePoolConfig := provider.MachinePoolFunc(generatedPoolName, namespace)
```

Comment:
Namespace is never passed into the func. It's a field of the `RKE2EncryptionKeyROtationTestSuite` struct, shouldn't it be `r.namespace?` since I'm assuming you've smoke tested this and it works, I'm probably missing something about Go here

Reply:
No you're right it probably needs to be updated to `r.namespace`.

Reply:
I lied, `namespace` is defined in `tests/v2/validation/provisioning/rke2/config.go` as `fleet-default`, which makes sense.